From f006ce0081d71b16392714c5a4df5bcd25fbb37d Mon Sep 17 00:00:00 2001
From: dnagasun <devamekalai.nagasundaram@intel.com>
Date: Fri, 1 Nov 2019 17:22:26 +0800
Subject: [PATCH] [VP] Fix Gpu hang decode

[Internal]
Platforms: Gen11
OS: Linux
Feature impact: CSC_Scal_Rot_Mirr
Resolves: HSD-1507492781
Related-to: N/A
Klocwork: N/A
IP Scan: PASS

Author  Chen, Jason K <jason.k.chen@intel.com>
Signed-off-by: dnagasun <devamekalai.nagasundaram@intel.com>

---
 .../agnostic/gen11/vp/hal/vphal_render_vebox_g11_base.cpp       | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/media_driver/agnostic/gen11/vp/hal/vphal_render_vebox_g11_base.cpp b/media_driver/agnostic/gen11/vp/hal/vphal_render_vebox_g11_base.cpp
index b6af1200..79a56d70 100644
--- a/media_driver/agnostic/gen11/vp/hal/vphal_render_vebox_g11_base.cpp
+++ b/media_driver/agnostic/gen11/vp/hal/vphal_render_vebox_g11_base.cpp
@@ -2215,7 +2215,7 @@ VPHAL_OUTPUT_PIPE_MODE VPHAL_VEBOX_STATE_G11_BASE::GetOutputPipe(
         pTarget->rcDst.bottom = pSrcSurface->rcDst.bottom;
 
         // Check if Vebox can be the output pipe again
-        bOutputPipeVeboxFeasible = IS_OUTPUT_PIPE_VEBOX_FEASIBLE(pVeboxState, pcRenderParams, pSrcSurface);
+        bOutputPipeVeboxFeasible = IS_OUTPUT_PIPE_VEBOX_FEASIBLE(pVeboxState, pcRenderParams, pSrcSurface) && !MEDIA_IS_SKU(pVeboxState->m_pSkuTable, FtrDisableVEBoxFeatures);
         if (bOutputPipeVeboxFeasible)
         {
             OutputPipe              = VPHAL_OUTPUT_PIPE_MODE_VEBOX;
-- 
2.23.0

